server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1927,7 +1928,8 @@ DiscoveryNode getNode() {
      * Else it configures the size to 80% of available capacity for a dedicated search node, if not explicitly defined.
      */
     private void initializeFileCache(Settings settings, CircuitBreaker circuitBreaker) throws IOException {
-        if (DiscoveryNode.isSearchNode(settings)) {
+        boolean isWritableRemoteIndexEnabled = FeatureFlags.isEnabled(FeatureFlags.WRITEABLE_REMOTE_INDEX_SETTING);
+        if (DiscoveryNode.isSearchNode(settings) || isWritableRemoteIndexEnabled) {
```

Comment:
With Composite Directory, FileCache will not be tied to Search roles. We should think about creating new roles specific to Warm as well to avoid any confusion with Search role (which is currently tied to Remote Snapshots only)